automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoApiClient.scala

```diff
@@ -223,33 +211,28 @@ object LeonardoApiClient {
     googleProject: GoogleProject,
     runtimeName: RuntimeName
   )(implicit client: Client[IO], authHeader: Authorization): IO[GetRuntimeResponseCopy] =
-    client.expect[GetRuntimeResponseCopy](
+    client.expectOr[GetRuntimeResponseCopy](
       Request[IO](
         method = Method.GET,
         headers = Headers.of(authHeader),
         uri = rootUri.withPath(s"/api/google/v1/runtimes/${googleProject.value}/${runtimeName.asString}")
       )
-    )
+    )(onError(s"Failed to get runtime ${googleProject.value}/${runtimeName.asString}"))
 
   def deleteRuntime(googleProject: GoogleProject,
                     runtimeName: RuntimeName,
                     deleteDisk: Boolean = true)(implicit client: Client[IO], authHeader: Authorization): IO[Unit] =
     client
-      .successful(
+      .expectOr[String](
         Request[IO](
           method = Method.DELETE,
           headers = Headers.of(authHeader),
           uri = rootUri
             .withPath(s"/api/google/v1/runtimes/${googleProject.value}/${runtimeName.asString}")
             .withQueryParam("deleteDisk", deleteDisk)
         )
-      )
-      .flatMap { success =>
-        if (success)
-          IO.unit
-        else
-          IO.raiseError(new RuntimeException(s"Fail to delete runtime ${googleProject.value}/${runtimeName.asString}"))
```

Comment:
I was annoyed all these http4s exceptions don't actually log the error response status code/body. So I improved the error handling and made it consistent.